Singularity/Singularity/platform/PlatformBlank.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Singularity.property;
+using Singularity.Resources;
+using Singularity.Units;
+
+namespace Singularity.platform
+{
+
+    internal class PlatformBlank : IDraw, IUpdate
+    {
+        private int mHealth;
+        private int mId;
+        private bool mIsBlueprint;
+        private readonly Action[] mActions;
+        private readonly Vector2 mPosition;
+        private readonly Texture2D mSpritesheet;
+        private readonly Dictionary<CUnit, Job> mAssignedUnits;
+        private List<IResources> mResources;
+        private Dictionary<IResources, int> mRequested;
+        private readonly Dictionary<IResources, int> mCost;
+
+
+
+        /// <summary>
+        /// Get the assigned Units of this platform.
+        /// </summary>
+        /// <returns> a list containing references of the units</returns>
+        public Dictionary<CUnit, Job> GetAssignedUnits()
+        {
+            return mAssignedUnits;
+        }
+
+        /// <summary>
+        /// Assign Units to this platform.
+        /// </summary>
+        /// <param name="unit">The unit to be assigned.</param>
+        /// <param name="job">The Job to be done by the unit</param>
+        public void AssignUnits(CUnit unit, Job job)
+        {
+            mAssignedUnits.Add(unit, job);
+        }
+
+        /// <summary>
+        /// Remove an Assigned Unit from the Assigned List.
+        /// </summary>
+        /// <param name="unit">The unit to unassign.</param>
+        public void UnAssignUnits(CUnit unit)
+        {
+            mAssignedUnits.Remove(unit);
+        }
+
+        /// <summary>
+        /// Get the Position of the platform as a 2dimensional vector.
+        /// </summary>
+        /// <returns>a Vector2 containing the position</returns>
+        public Vector2 GetPosition()
+        {
+            return mPosition;
+        }
+
+        /// <summary>
+        /// Get the special actions you can perform on this platform.
+        /// </summary>
+        /// <returns> an array with the available actions.</returns>
+        public Action[] GetActions()
+        {
+            return mActions;
+        }
+
+        /// <summary>
+        /// Perform the given action on the platform.
+        /// </summary>
+        /// <param name="action"> The action to be performed </param>
+        /// <returns> true if it was succesfull</returns>
+        public bool DoAction(Action action)
```

Comment:
Why is it a bool here?

Reply:
I assume at some point back then I decided it was a good idea to return true if the action was executed successfully and forgot to change it. And now that I think about it it could be useful at some actions.